### PR TITLE
NON-253: Forbid selecting someone who already has an open non-association with key prisoner

### PR DESCRIPTION
**Pull request by ushkarev**

### CODE REVIEW

File: server/routes/prisonerSearch.ts [hunk 1]
```diff
@@ -122,6 +124,19 @@ export default function prisonerSearchRoutes(service: Services): Router {
...
+          const api = new NonAssociationsApi(systemToken)
+          const openNonAssociations = await api.listNonAssociationsBetween([
+            prisonerNumber,
+            ...response.content.map(somePrisoner => somePrisoner.prisonerNumber),
+          ])
+          openNonAssociations?.forEach(nonAssociation => {
+            if (nonAssociation.firstPrisonerNumber === prisonerNumber) {
+              openNonAssociationsMap.set(nonAssociation.secondPrisonerNumber, nonAssociation.id)
+            } else if (nonAssociation.secondPrisonerNumber === prisonerNumber) {
+              openNonAssociationsMap.set(nonAssociation.firstPrisonerNumber, nonAssociation.id)
```
Comment (xoen) [inline server/routes/prisonerSearch.ts:137]: What about something like?

```JavaScript
let otherPrisonerNumber = [nonAssociation.firstPrisonerNumber, nonAssociation.secondPrisonerNumber].find(pn => pn !== prisonerNumber)
openNonAssociationsMap.set(otherPrisonerNumber, nonAssociation.id)
```

Maybe too clever. Or maybe use ternary operator 🤔 

```JavaScript
let otherPrisonerNumber = nonAssociation.firstPrisonerNumber === prisonerNumber ? nonAssociation.secondPrisonerNumber : nonAssociation.firstPrisonerNumber
openNonAssociationsMap.set(otherPrisonerNumber, nonAssociation.id)
```

Doesn't really matter, just a couple of ideas.

**Review by xoen — APPROVED** [review]

File: server/routes/prisonerSearch.ts (hunk 1 above)
Comment (ushkarev) [reply server/routes/prisonerSearch.ts:137]: only one `if` branch so not tooo bad. my first version added both and then deleted the key prisoner at the end. i'm happy with your second version…

**Review by xoen — APPROVED** [review]

**Review by xoen — APPROVED** [review]

File: server/routes/prisonerSearch.ts (hunk 1 above)
Comment (ushkarev) [reply server/routes/prisonerSearch.ts:137]: oopsy. i forgot that the list of non-associations here might not include the key prisoner at all. thankfully a test caught that 💪

Comment (xoen) [reply server/routes/prisonerSearch.ts:137]: Sorry, my bad. This is not an `if`/`else` so the ternary operator doesn't make sense in this case.
Also the non-associations returned from the API (between endpoint) may be between 2 other prisoners, e.g. not involving the `prisonerNumber` one, so what I suggested didn't make sense in this context, soz :)

**Pull request merged**